spring-cloud-stream-binder-kafka-streams/src/main/java/org/springframework/cloud/stream/binder/kafka/streams/KafkaStreamsStreamListenerSetupMethodOrchestrator.java

```diff
@@ -288,8 +296,48 @@ else if (parameterType.isAssignableFrom(KTable.class)) {
 						.withValueSerde(v));
 	}
 
-	private KStream<?, ?> getkStream(String inboundName, BindingProperties bindingProperties, StreamsBuilder streamsBuilder,
-									Serde<?> keySerde, Serde<?> valueSerde) {
+	private StoreBuilder buildStateStore(KafkaStreamsStateStoreProperties spec) throws Exception {
+		try {
+			Serde<?> keySerde = this.keyValueSerdeResolver.getStateStoreKeySerde(spec.getKeySerdeString());
+			Serde<?> valueSerde = this.keyValueSerdeResolver.getStateStoreValueSerde(spec.getValueSerdeString());
+			StoreBuilder builder = null;
+			switch (spec.getType()) {
+				case KEYVALUE:
+					builder = Stores.keyValueStoreBuilder(Stores.persistentKeyValueStore(spec.getName()), keySerde, valueSerde);
+					break;
+				case WINDOW:
+					builder = Stores.windowStoreBuilder(Stores.persistentWindowStore(spec.getName(), spec.getRetention(), 3, spec.getLength(), false),
+							keySerde,
+							valueSerde);
+					break;
+				case SESSION:
+					builder = Stores.sessionStoreBuilder(Stores.persistentSessionStore(spec.getName(), spec.getRetention()), keySerde, valueSerde);
+					break;
+				default:
+					throw new UnsupportedOperationException("state store type (" + spec.getType() + ") is not supported!");
+			}
+			if (spec.isCacheEnabled())
+				builder = builder.withCachingEnabled();
+			if (spec.isLoggingDisabled())
+				builder = builder.withLoggingDisabled();
+
+			return builder;
+
+		}catch (Exception e) {
+			LOG.error("failed to build state store exception : " + e);
+			throw e;
+		}
+	}
+
+
+	private KStream<?, ?> getkStream(String inboundName, KafkaStreamsStateStoreProperties storeSpec, BindingProperties bindingProperties, StreamsBuilder streamsBuilder,
+									Serde<?> keySerde, Serde<?> valueSerde) throws Exception {
+		if (storeSpec != null) {
+			StoreBuilder storeBuilder = buildStateStore(storeSpec);
+			Assert.notNull(storeBuilder);
```

Comment:
The assertion should not be used here. The `buildStateStore` is a private method service only functionality requested here, so the possibility of a null should be encapsulated in `buildStateStore` method.